analyses/snv-callers/util/tmb_functions.R

```diff
@@ -72,92 +74,74 @@ snv_ranges_filter <- function(maf_df, keep_ranges = NULL, bp_window = 0) {
   #   maf_df: maf data.frame that has been turned into a data.frame, has had
   #           the experimental_stategy column added from the metadata (can be
   #           done with the `set_up_maf` function) and has WXS mutations filtered
   #           using `wxs_bed_filter` function (If the situation calls for it).
-  #   bed_wgs: BED file path with genome ranges to be used for filtering and 
-  #            genome size denominator for WGS samples
-  #   bed_wxs: BED file path genome ranges to be used for filtering and genome 
-  #            size denominator for WXS samples
+  #   bed_df: BED formatted data.frame of the target regions to use for TMB
+  #           calculations.
   #
   # Returns:
-  # A sample-wise data.frame with Tumor Mutation Burden statistics calculated
-  # using the given WGS and WXS sizes.
-  
-  # Read in the BED files we need
-  bed_wgs <- readr::read_tsv(bed_wgs, col_names = FALSE)
-  bed_wxs <- readr::read_tsv(bed_wxs, col_names = FALSE)
-  
+  # A calculated genome size, and TMB for the given Tumor_Sample_Barcode,
+  # returned as a single row data.frame. `experimental_strategy`, `short_histology`
+  # columns are carried along.
+
   # Sum up genome sizes
-  wgs_size <- sum(bed_wgs[, 3] - bed_wgs[, 2])
-  wxs_size <- sum(bed_wxs[, 3] - bed_wxs[, 2])
-  
-  # Don't want integers per se
-  wgs_size <- as.numeric(wgs_size)
-  wxs_size <- as.numeric(wxs_size)
-  
-  # For WXS samples, filter out mutations that are outside of these coding regions.
-  filt_wxs_maf_df <- snv_ranges_filter(dplyr::filter(
-    maf_df,
-    experimental_strategy == "WXS"
-  ),
-  keep_ranges = bed_wxs
-  )
-  
-  # For WXS samples, filter out mutations that are outside of these coding regions.
-  filt_wgs_maf_df <- snv_ranges_filter(dplyr::filter(
-    maf_df,
-    experimental_strategy == "WGS"
-  ),
-  keep_ranges = bed_wgs
-  )
-  
-  # Bind the filtered WXS sample rows back to the WGS samples
-  filt_maf_df <- dplyr::bind_rows(filt_wxs_maf_df, filt_wgs_maf_df)
-  
+  bed_size <- as.numeric(sum(bed_ranges@ranges@width))
+
+  # Filter to only the sample's mutations
+  sample_maf_df <- maf_df %>%
+    dplyr::filter(Tumor_Sample_Barcode == tumor_sample_barcode)
+
+  # Filter out mutations that are outside of these coding regions.
+  filt_maf_df <- snv_ranges_filter(sample_maf_df, keep_ranges = bed_ranges)
+
   # Make a genome size variable
-  tmb <- maf_df %>%
-    dplyr::mutate(genome_size = dplyr::recode(experimental_strategy,
-                                              "WGS" = wgs_size,
-                                              "WXS" = wxs_size
-    )) %>%
+  tmb <- sample_maf_df %>%
     dplyr::group_by(
-      Tumor_Sample_Barcode, experimental_strategy, genome_size,
+      Tumor_Sample_Barcode = tumor_sample_barcode,
+      experimental_strategy,
       short_histology
     ) %>%
     # Count number of mutations for that sample
     dplyr::summarize(mutation_count = dplyr::n()) %>%
     # Calculate TMB
```

Comment:
You should be able do all the columns within the summarize statement; no need for mutate after:
```suggestion
    dplyr::summarize(
      mutation_count = dplyr::n(),
      region_size = bed_size,
      tmb = mutation_count / (region_size / 1000000)
     )
```